src/features/machines/swapIntentMachine.ts

```diff
@@ -588,3 +605,37 @@ async function verifyWalletSignature(
       throw new Error("exhaustive check failed")
   }
 }
+
+export function calcOperationAmountOut(
+  operation: IntentOperationParams
+): bigint {
+  if (operation.type === "swap") {
+    return operation.quote.totalAmountOut
+  }
+
+  return calcWithdrawAmount(
+    operation.quote,
+    operation.nep141Storage,
+    operation.directWithdrawalAmount
+  )
+}
+
+export function calcWithdrawAmount(
+  swapQuote: AggregatedQuote | null,
+  nep141Storage: NEP141StorageRequirement | null,
+  directWithdrawalAmount: bigint
+): bigint {
+  const gotFromSwap = swapQuote?.totalAmountOut ?? 0n
+
+  let spentOnStorage = 0n
+  if (nep141Storage != null) {
+    if (nep141Storage.type === "no_swap_needed") {
+      // Assume that token out is NEAR/wNEAR, so we can just use the required storage
+      spentOnStorage = nep141Storage.requiredStorageNEAR
```

Comment:
I didn't get how you do the calculation of requiredStorageNEAR. Are you getting this from the solver?

Reply:
I get it from chain in `getNEP141StorageRequired`. Then it is used in `determineNEP141StorageRequirement` to know if swap needed or not.